finality-aleph/src/aggregation/mod.rs

```diff
@@ -81,17 +103,24 @@ where
         }
     }
 
-    pub async fn start_aggregation(&mut self, h: BlockHash) {
+    pub async fn start_aggregation(&mut self, h: EitherHash) {
+        use EitherHash::*;
         match &mut self.agg {
             EitherAggregator::Current(agg) => agg.start_aggregation(h).await,
-            EitherAggregator::Legacy(agg) => agg.start_aggregation(h).await,
+            EitherAggregator::Legacy(agg) => match h {
+                Block(h) => agg.start_aggregation(h).await,
+                Performance(_) => { /* should never happen, but ignoring is fine */ }
```

Comment:
Maybe panic here? Like Jon says here https://youtu.be/qfknfCsICUM?si=HUOv2aXBnkS1M6JE&t=342 . If we are truly right this never happens, this is an unrecoverable error, which means we should panic here

Reply:
It should never happen, but it's not really an unrecoverable error, since ignoring it is always fine. That's why there is an `unimplemented!` in the other similar spot currently, that one is an actual problem _if_ it happens, even though it cannot.